src/openforms/formio/components/vanilla.py

```diff
@@ -348,3 +349,28 @@ def rewrite_for_request(component: ContentComponent, request: Request):
            security risk.
         """
         component["html"] = post_process_html(component["html"], request)
+
+
+@register("editgrid")
+class EditGrid(BasePlugin[EditGridComponent]):
+    def build_serializer_field(
+        self, component: EditGridComponent
+    ) -> serializers.ListField:
+        validate = component.get("validate", {})
+        required = validate.get("required", False)
+        nested = build_serializer(
+            components=component.get("components", []),
+            # XXX: check out type annotations here, there's some co/contra variance
+            # in play
+            register=self.registry,
```

Comment:
This can be fixed by adding `registry: "ComponentRegistry"` as a type annotation on `BasePlugin` (currently `self.registry` refers to `BaseRegistry`, and `build_serializer` expects `ComponentRegistry` -- inheritance is `BaseRegistry > ComponentRegistry`).

Reply:
Ah yes saw it fixed the XXX comment, but created this new error